From 9be93a21e8d466df8755a7670008a00942af31e2 Mon Sep 17 00:00:00 2001
From: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
Date: Wed, 20 Feb 2019 10:20:09 +0200
Subject: [PATCH] iommu: Fix iommu-map entry in fsl-mc node from dts

The iommu-map entry from the fsl-mc node is updated by valid
stream-ids by u-boot version 2017.12. The iommu driver contains
a method to update the iommu-map entry, but it is not invoked
from the source code which configures the ls2084abbmini board.
This leads to faulty initialization of iommu by linux kernel version 4.14.
We solve this by invoking 'fdt_fsl_mc_fixup_iommu_map_entry'
method which updates the iommu-map entry whithin the board setup.

Signed-off-by: Ciprian Costea <ciprianmarian.costea@nxp.com>
---
 board/freescale/ls2084abbmini/ls2084abbmini.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/board/freescale/ls2084abbmini/ls2084abbmini.c b/board/freescale/ls2084abbmini/ls2084abbmini.c
index 6e88a01..00e0d3e 100644
--- a/board/freescale/ls2084abbmini/ls2084abbmini.c
+++ b/board/freescale/ls2084abbmini/ls2084abbmini.c
@@ -294,6 +294,8 @@ int ft_board_setup(void *blob, bd_t *bd)
 
 	fdt_fixup_memory_banks(blob, base, size, 2);
 
+   fdt_fsl_mc_fixup_iommu_map_entry(blob);
+
 	fsl_fdt_fixup_dr_usb(blob, bd);
 
 #ifdef CONFIG_FSL_MC_ENET
-- 
1.9.1

